src/scheduler/scheduler.controller.js

```diff
@@ -0,0 +1,149 @@
+            LoadingModal.close();
+        });
+
+        JobsService.setListener($scope);
+        JobsService.fetchJobs({});
+
+        $scope.changePageTo = changePageTo;
+
+        function changePageTo(page){
+            JobsService.setParam("page", page);
+            JobsService.fetchJobs();
+        }
+
+        var searchUpdateTimeout;
+        function updateSearch(searchData){
+
+            if(searchUpdateTimeout){
+                clearTimeout(searchUpdateTimeout);
+                searchUpdateTimeout=null;
+            }
+            searchUpdateTimeout = setTimeout(function() {
+                JobsService.setParam("name", searchData.name);
+                JobsService.setParam("activity", searchData.activity);
+                JobsService.setParam("status", searchData.status);
+                JobsService.setParam("timeFrom", searchData.timeFrom);
+                JobsService.setParam("timeTo", searchData.timeTo);
+                JobsService.fetchJobs();
+            }, 1000);
+        }
+
+        $scope.getDetails = function(job) {
+            if ($scope.jobDetails[job.name] !== undefined) {
+            } else {
+                if (!job.uiDefined) {
+                    JobsService.getDetails(job, function(data) {
+                        $scope.jobDetails[job.name] = data;
+                    });
+                }
+            }
+        };
+
+        $scope.updateJob = function(job, updated) {
+            job.activity = updated.activity;
+            job.status = updated.status;
+            job.name = updated.name;
+            job.group = updated.group;
+            job.startDate = updated.startDate;
+            job.nextFireDate = updated.nextFireDate;
+            job.endDate = updated.endDate;
+            job.jobType = updated.jobType;
+            job.info = updated.info;
+            job.uiDefined = updated.uiDefined;
+        };
+
+        $scope.pauseJob = function(job) {
+            ModalFactory.showConfirm("scheduler.confirm.pause", "scheduler.confirm", function(response) {
+                if (response) {
+                    LoadingModal.open();
+                    JobsService.pauseJob(job, function(updated) {
+                        $scope.updateJob(job, updated);
+                        LoadingModal.close();
+                    });
+                }
+            });
+        };
+
+        $scope.resumeJob = function(job) {
+            ModalFactory.showConfirm("scheduler.confirm.resume", "scheduler.confirm", function(response) {
+                if (response) {
+                    LoadingModal.open();
+                    JobsService.resumeJob(job, function(updated) {
+                       $scope.updateJob(job, updated);
+                       LoadingModal.close();
+                    });
+                }
+            });
+        };
+
+        $scope.newJob = function() {
+            LoadingModal.open();
+        };
+
+        $scope.editJob = function(job) {
```

Comment:
I don't think this function needs to exist either, right? Ideally we could pass the current job ID through UI-Router instead of setting the JobState in the service?